src/partition/repart.c

```diff
@@ -1480,6 +1562,27 @@ static int context_read_definitions(
                 context->n_partitions++;
         }
 
+        /* Check that each configured verity hash partition has a matching data partition. */
+
+        LIST_FOREACH(partitions, p, context->partitions) {
+                Partition *dp;
+
+                if (p->verity == VERITY_DATA)
+                        continue;
+
+                dp = find_verity_data_partition(context, p);
+                if (!dp)
+                        return log_syntax(NULL, LOG_ERR, p->definition_path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                          "Missing verity data partition for verity partition from %s",
+                                          p->definition_path);
+
+                p->data_partition = dp;
+
+                if (dp->encrypt != ENCRYPT_OFF)
+                        return log_syntax(NULL, LOG_ERR, dp->definition_path, 1, SYNTHETIC_ERRNO(EINVAL),
+                                          "A verity data partition cannot be encrypted");
```

Comment:
i would say "is not supported" rather than "cannot". I mean, it obviously *can* be encrypted, we just decide not to bother with it, since it's pretty useless...